saleor/graphql/webhook/mutations.py

```diff
@@ -301,3 +308,69 @@ def perform_mutation(cls, _root, info, **data):
         manager = get_plugin_manager_promise(info.context).get()
         manager.event_delivery_retry(delivery)
         return EventDeliveryRetry(delivery=delivery)
+
+
+class WebhookDryRunInput(graphene.InputObjectType):
+    object_id = graphene.ID(
+        description="The ID of an object to serialize.", required=True
+    )
+    query = graphene.String(
+        description="The subscription query that defines the webhook event and its "
+        "payload.",
+        required=True,
+    )
+
+
+class WebhookDryRun(BaseMutation):
+    payload = JSONString(
+        description="JSON payload, that would be sent out to webhook's target URL."
+    )
+
+    class Arguments:
+        input = WebhookDryRunInput(
+            description="Fields required to perform dry run of a webhook event.",
+            required=True,
+        )
```

Comment:
Hmm maybe actually we don't. It will be needed for the mutation `webhookTrigger` for sure to know for which webhook to send the request (target URL etc.). I would only check what happens when running dry-run using a query with fields you that don't have permission to fetch. E.g. the app might have only MANAGE_ORDERS permission, but in the query you might ask for some restricted product-level fields that require MANAGE_PRODUCT. I would check if the payload generation function handles that and returns the information that you cannot query those fields.